File: python/series/series.py
def slices(series, length):
    if len(series) == 0:
        raise ValueError("Series must have positive length.")
    if length < 1:
        raise ValueError("Slice length must be positive.")
    if length > len(series):
        raise ValueError("Slice length must be at most the series length.")

    return [series[x:x+length] for x in range(len(series) - length + 1)]

File: csharp/series/Series.cs
using System.Collections.Generic;

public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (sliceLength > numbers.Length)
            throw new System.ArgumentException($"sliceLength ({sliceLength}) larger than numbers string.");

        if (sliceLength < 1)
            throw new System.ArgumentException($"sliceLength must be positive: given {sliceLength}");

        var slices = new List<string>();
        int loopEnd = numbers.Length - sliceLength;
        for (int i = 0; i <= loopEnd; i++)
            slices.Add(numbers.Substring(i, sliceLength));

        return slices.ToArray();
    }
}